Riot/Modules/Pills/PillAttachmentView.swift

```diff
@@ -0,0 +1,112 @@
+        static let avatarSideLength: CGFloat = 16.0
+        static let pillBackgroundHeight: CGFloat = avatarSideLength + 2 * commonVerticalMargin
+        static let displaynameLabelLeading: CGFloat = avatarSideLength + 2 * commonHorizontalMargin
+        static let pillHeight: CGFloat = pillBackgroundHeight + 2 * commonVerticalMargin
+        static let displaynameLabelTrailing: CGFloat = 2 * commonHorizontalMargin
+        static let totalWidthWithoutLabel: CGFloat = displaynameLabelLeading + displaynameLabelTrailing
+    }
+
+    // MARK: - Init
+    /// Create a Mention Pill view for given room member.
+    ///
+    /// - Parameters:
+    ///   - roomMember: the room member
+    ///   - isHighlighted: whether this pill should be highlighted
+    convenience init(withRoomMember roomMember: MXRoomMember, isHighlighted: Bool) {
+        self.init(frame: CGRect(origin: CGPoint(x: 0.0, y: 0.0),
+                                size: Self.size(forRoomMember: roomMember)))
+        let label = UILabel(frame: .zero)
+        label.text = roomMember.displayname.count > 0 ? roomMember.displayname : roomMember.userId
+        label.font = Constants.pillLabelFont
+        label.textColor = isHighlighted ? ThemeService.shared().theme.baseTextPrimaryColor : ThemeService.shared().theme.textPrimaryColor
+        let labelSize = label.sizeThatFits(CGSize(width: CGFloat.greatestFiniteMagnitude,
+                                                  height: CGFloat.greatestFiniteMagnitude))
+        label.frame = CGRect(x: Constants.displaynameLabelLeading,
+                             y: 0,
+                             width: labelSize.width,
+                             height: Constants.pillBackgroundHeight)
+
+        let pillBackgroundView = UIView(frame: CGRect(x: 0,
+                                        y: Constants.commonVerticalMargin,
+                                        width: labelSize.width + Constants.totalWidthWithoutLabel,
+                                        height: Constants.pillBackgroundHeight))
+
+        let imageView = MXKImageView(frame: CGRect(x: Constants.commonHorizontalMargin,
+                                                   y: Constants.commonVerticalMargin,
+                                                   width: Constants.avatarSideLength,
+                                                   height: Constants.avatarSideLength))
+        imageView.setImageURI(roomMember.avatarUrl,
+                              withType: nil,
+                              andImageOrientation: .up,
+                              toFitViewSize: imageView.frame.size,
+                              with: MXThumbnailingMethodCrop,
+                              previewImage: Asset.Images.userIcon.image,
+                              // Pills rely only on cached images since `MXKImageView` image loading
+                              // is not handled properly for a `NSTextAttachment` view.
```

Comment:
(this might require further tests to see if we can do something)

The issue here is not getting the manager, I had that no problem from context with session. 

The issue is that the attachment view doesn't actually redraw (maybe we can force it somehow)